util/src/ifaces/ffi/windows/mod.rs

```diff
@@ -316,6 +320,44 @@ unsafe fn local_ifaces_with_buffer(buffer: &mut Vec<u8>) -> io::Result<()> {
     }
 }
 
+/// Convert wide-char(16bit) pointer to OsString.
+/// Note: When the string length exceeds buffer_size, it will return empty.
+fn windows_pwchar_to_string(ptr: *const wchar_t, buffer_size: usize) -> OsString {
+    if ptr.is_null() || buffer_size == 0 {
+        return OsString::new();
+    }
+
+    extern "C" {
+        /// Provided by libc or compiler_builtins.
+        /// determine the length of a fixed-size wide-character string.
+        ///
+        /// In windows, wchar_t is 16bit, in GNU LibC it is 32bit,
+        /// So in Windows, the length of both wchar_t and WCHAR is equal.
+        /// See https://learn.microsoft.com/en-us/cpp/cpp/char-wchar-t-char16-t-char32-t
+        fn wcsnlen(buf: *const wchar_t, size: size_t) -> usize;
+    }
+
+    // # SAFETY
+    // Caller has provided a pointer of valid C String, And confirm the maximum length of the buffer.
```

Comment:
Given this is placing requirements on the caller shouldn't the whole function be `unsfafe` it seems possible to misuse this function from safe code but trigger undefined behaviour. WDYT?